actors/base/src/main/scala/com/waz/provision/DeviceActor.scala

```
@@ -463,6 +463,15 @@ class DeviceActor(val deviceName: String,
         Successful
       }
 
+    case UpdateProfileUserName(userName) =>
+      whenSelfLoaded { self =>
+        self.setUsername(userName, new CredentialsUpdateListener {
+          override def onUpdateFailed(code: Int, message: String, label: String): Unit = ()
```

You might want to return a failure message with the code and reason to help you debug, but up to you